File: split_files.js
const fs = require('fs');

function splitProject() {
  const html = fs.readFileSync('index.html', 'utf-8');
  
  const cssMatch = html.match(/<style>([\s\S]*?)<\/style>/);
  const jsMatch = html.match(/<script>\s*(\(function\(\)\{[\s\S]*?\}\)\(\);)\s*<\/script>/);

  if (cssMatch && jsMatch) {
    if (!fs.existsSync('css')) fs.mkdirSync('css');
    if (!fs.existsSync('js')) fs.mkdirSync('js');

    fs.writeFileSync('css/style.css', cssMatch[1].trim(), 'utf-8');
    fs.writeFileSync('js/app.js', jsMatch[1].trim(), 'utf-8');

    let newHtml = html.replace(/<style>[\s\S]*?<\/style>/, '<link rel="stylesheet" href="css/style.css">');
    newHtml = newHtml.replace(/<script>[\s\S]*?<\/script>/, '<script src="js/app.js"></script>');

    fs.writeFileSync('index.html', newHtml, 'utf-8');
    console.log("Arquivos divididos com sucesso!");
  } else {
    console.error("Erro: Não foi possível encontrar as tags style e script.");
  }
}

splitProject();

File: split_files.py
import os
import re

def split_project():
    with open('index.html', 'r', encoding='utf-8') as f:
        html = f.read()

    css_match = re.search(r'<style>(.*?)</style>', html, re.DOTALL)
    js_match = re.search(r'<script>(.*?)</script>', html, re.DOTALL)

    if css_match and js_match:
        os.makedirs('css', exist_ok=True)
        os.makedirs('js', exist_ok=True)

        with open('css/style.css', 'w', encoding='utf-8') as f:
            f.write(css_match.group(1).strip())

        with open('js/app.js', 'w', encoding='utf-8') as f:
            f.write(js_match.group(1).strip())

        new_html = html.replace(css_match.group(0), '<link rel="stylesheet" href="css/style.css">')
        new_html = new_html.replace(js_match.group(0), '<script src="js/app.js"></script>')

        with open('index.html', 'w', encoding='utf-8') as f:
            f.write(new_html)
            
        print("Arquivos divididos com sucesso!")
    else:
        print("Erro: Não foi possível encontrar as tags style e script.")

if __name__ == "__main__":
    split_project()
